src/promptflow/promptflow/_sdk/operations/_tool_operations.py

```diff
@@ -189,3 +419,55 @@ def list(
             tools = {"package": {}, "code": {}}
         tools["package"].update(package_tools)
         return tools
+
+    @monitor_operation(activity_name="pf.tools.validate", activity_type=ActivityType.PUBLICAPI)
+    def validate(
+        self, source: Union[str, callable, PathLike], *, raise_error: bool = False, **kwargs
+    ) -> ValidationResult:
```

Comment:
ValidationResult should be public exposed？

Reply:
Yes, ValidationResult  needs to be exposed to user